src/renderer/Material.ts

```diff
@@ -0,0 +1,66 @@
+import { vec3 } from 'gl-matrix';
+import { Bakeable, Color } from '../calder';
+import { RGBColor } from '../colors/RGBColor';
+
+/**
+ * Required parameters when defining a `Material`.
+ */
+export type MaterialParams = {
+    color: Color;
+    shininess: number;
+};
+
+/**
+ * The representation of a `Material` when passed to the shader.
+ */
+export type BakedMaterial = {
+    materialColor: vec3;
+    materialShininess: number;
+};
+
+/**
+ * A simple representation of the material of an object in a scene.
+ */
+export class Material implements Bakeable {
+    public readonly materialColor: vec3;
+    public readonly materialShininess: number;
+
+    /**
+     * Instantiates a new Material.
+     *
+     * @class Material
+     * @constructor
+     * @param {MaterialParams} params The named parameters for this material.
+     */
+    constructor(params: MaterialParams) {
+        this.materialColor = params.color.asVec();
+        this.materialShininess = params.shininess;
+    }
+
+    /**
+     * Instantiates a new Material.
+     *
+     * @param {MaterialParams} params The named parameters for this material.
+     * @returns {Material} A new material.
+     */
+    public static create(params: MaterialParams): Material {
```

Comment:
I think it's ok not having that be part of the interface and just a convention for API consistency

Reply:
Hmm, I kinda like having it in _an_ interface (not necessarily this one) because it's explicit about how we instantiate things across the classes we expose to the end-user. I realize it's a huge nit, and we (the Calder team) know to enforce this convention, but I think having it be explicit in an interface is more apparent to external people viewing the code, and the compiler will complain if we don't implement all the methods in the interface for some class which implements it.

Reply:
We can do this in another PR (I can do it), but I like the idea of having more interfaces to communicate _intent_ about what we're doing, and what we're exposing.